core/process.py

```diff
@@ -1462,11 +1476,13 @@ def WaitForOne(self, eintr_retry):
       if term_sig == signal_.SIGINT:
         print('')
 
+      self.job_state.NotifyDone(pid)
       proc.WhenDone(pid, status)
 
     elif WIFEXITED(status):
       status = WEXITSTATUS(status)
       #log('exit status: %s', status)
+      self.job_state.NotifyDone(pid)
```

Comment:
I think maybe we should rename these

self.job_state.WhenDone()

self.job_state.WhenContinued()

self.job_state.WhenStopped()

What do you think?

Reply:
Sure, that's what `Process` et. al. use. `Notify*` suggests some kind of callback, but everything here is synchronous.